.github/workflows/invalidatePRs.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+# GitHub workflow handles invalidating Pull Requests after 3 hours.
+#
+# It will only invalidate open, non-draft pull requests and mark each status
```

Comment:
Can you add a bit more description here about the possible cases. Similar to my response to Grigas, we should write out how it works in those explicit cases.

Reply:
Added some description and edited some comments as per your comment. Let me know what you think of the description (possible cases) I added. I am not sure if I accounted all the explicit cases or not.